File: HDOJ/CCPC16Online/src/1001.cc
/*************************************************************************
  >         File: 1001.cc
  >       Author: Riho.Yoshioka
  >         Mail: riho.yoshioka@yandex.com
  > Created Time: Sun 14 Aug 2016 02:36:57 PM CST
*************************************************************************/
#include <bits/stdc++.h>
typedef long long LL;
using namespace std;

char s[12345678];

int go(char *s, int x) {
  int n = strlen(s);
  int ans = 0;
  for (int i = 0; i < n; i++) {
    ans = 10 * ans + s[i] - '0';
    ans %= x;
  }
  return ans;
}

int main() {
  ios::sync_with_stdio(0);
  int ca = 1;
  while (scanf("%s", s) > 0) {
    cout << "Case #" << ca++ << ": ";
    if (go(s, 73) == 0 && go(s, 137) == 0) {
      cout << "YES" << endl;
    }
    else {
      cout << "NO" << endl;
    }
  }
  return 0;
}

File: HDOJ/CCPC16Online/src/1001.java
/*************************************************************************
  >         File: Main.java
  >       Author: Riho.Yoshioka
  >         Mail: riho.yoshioka@yandex.com
  > Created Time: Sun 14 Aug 2016 02:25:38 PM CST
*************************************************************************/
import java.io.*;
import java.util.*;
import java.math.*;

public class Main {
  public static void main(String[] args) {
    InputStream inputStream = System.in;
    OutputStream outputStream = System.out;
    InputReader in = new InputReader(inputStream);
    PrintWriter out = new PrintWriter(outputStream);
    
    TaskA solver = new TaskA();
    solver.solve(1, in, out);
    out.close();
  }

  static class TaskA {
    public void solve(int testNumber, InputReader in, PrintWriter out) {
      BigInteger A = new BigInteger("73"), B = new BigInteger("137");
      int ca = 1;
      Scanner s = new Scanner(System.in);
      while (s.hasNext()) {
        String now = s.next();
        BigInteger a = new BigInteger(now);
        out.print("Case #" + ca++ + ": ");
        if (a.mod(A).equals(BigInteger.ZERO) && a.mod(B).equals(BigInteger.ZERO)) {
          out.println("YES");
        }
        else {
          out.println("NO");
        }
      }
    }
  }

  static class InputReader {
    public BufferedReader reader;
    public StringTokenizer tokenizer;

    public InputReader(InputStream stream) {
      reader = new BufferedReader(new InputStreamReader(stream), 32768);
      tokenizer = null;
    }

    public String next() {
      while (tokenizer == null || tokenizer.hasMoreTokens()) {
        try {
          tokenizer = new StringTokenizer(reader.readLine());
        } catch (IOException e) {
          throw new RuntimeException(e);
        }
      }
      if (!tokenizer.hasMoreTokens()) return null;
      return tokenizer.nextToken();
    }

    public int nextInt() {
      return Integer.parseInt(next());
    }

    public long nextLong() {
      return Long.parseLong(next());
    }

    public double nextDouble() {
      return Double.parseDouble(next());
    }
  }
}
